### PR TITLE
Clarify gpg verification

### PR DESCRIPTION
**Pull request by hasufell**
Some docker image developers not understanding what a hash function is have previously told me my installer has to gpg-verify all the tarballs instead of their checksums.

Of course that's nonsense, because then the internet, all distros (including ubuntu) and even blockchains in this world would be broken.

Adding this as a note here should silence this.

Thanks.

### CODE REVIEW

**Comment by hasufell** [comment]
> However, to properly do it "justice" it ends up getting a bit wordy -- perhaps there's a decent guide or article someone has written elsewhere that we could link to instead of including all this detail here?
> 
> (Otherwise I'd probably prefer to punt this addition to be part of #6282 instead so we can have something like a dedicated file to describe low-level details like this better and keep the high-level overview short.)

Could you clarify whether you're interested in me improving this PR? And if so what exactly do you need?

I don't know of any guide explaining GPG+hash use. It's somewhat basic cryptography knowledge.

**Comment by tianon** [comment]
Honestly, I don't have a really strong opinion on the best way to incorporate this yet. I was hoping we could discuss it a bit more and help tip the scales, but it sounds like you're not up for that (which is totally fair).

Given that, I'm going to add this to the backlog, and we'll come back to it later; thanks for the suggestion!
